installutils/helminstall/internal/memory_backed_rest_client_getter.go

```diff
@@ -0,0 +1,90 @@
+package internal
+
+import (
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	diskcached "k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	defaultCacheDir = filepath.Join(clientcmd.RecommendedConfigDir, "http-cache")
+
+	// overlyCautiousIllegalFileCharacters matches characters that *might* not be supported.  Windows is really restrictive, so this is really restrictive
+	overlyCautiousIllegalFileCharacters = regexp.MustCompile(`[^(\w/\.)]`)
+
+	removeProtocol = regexp.MustCompile("http[s]?:/{2}")
+)
+
+func NewInMemoryRESTClientGetter(rawConfig clientcmd.ClientConfig) genericclioptions.RESTClientGetter {
+	return &inMemoryRESTClientGetter{rawConfig: rawConfig}
+}
+
+type inMemoryRESTClientGetter struct {
+	rawConfig clientcmd.ClientConfig
+}
+
+func (i *inMemoryRESTClientGetter) ToRESTConfig() (*rest.Config, error) {
+	config, err := i.rawConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+func (i *inMemoryRESTClientGetter) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	cfg, err := i.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	return ToDiscoveryClient(cfg)
+}
+
+func (i *inMemoryRESTClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
+	cfg, err := i.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+	return ToRESTMapper(cfg)
+}
+
+func (i *inMemoryRESTClientGetter) ToRawKubeConfigLoader() clientcmd.ClientConfig {
+	return i.rawConfig
+}
+
+//
+// files below this comment were shamelessly and brazenly stolen from the Helm codebase
```

Comment:
if so, could be helpful to link to the source

Reply:
@GrahamGoudeau do you recall where in the Helm codebase you copy pasta-ed this from?